### PR TITLE
Small cleanup, envy.home support and throwaway envs

### LINKED ISSUE #16: throwaway environments
Sessions with an explicitly blank environment name will produce a throwaway environment. It will be named `env.$(date +%s)` and the image will be deleted when the session ends. 
#### SSH example

```
$ ssh progrium+@envy.host
```
#### HTTP example

```
https://envy.host/u/progrium/-
```

### PR DESCRIPTION
**Pull request by fgrehm**
This provides just a little bit cleanup that I think will make other people lives easier and it should be able to close GH-16 and GH-14.

The part where it checks for the throwaway environment is a bit messed up but it is the best I could do with my current bash knowledge. I'm happy to refactor it if anyone knows a better way of doing that :blush: 

_Sorry for doing too much in a single PR, I'm cool to extract them into smaller ones :grin:_

### CODE REVIEW

**Comment by progrium** [comment]
This is really great. Because the project is so young and I have really high bash standards, I might refactor this before merging. But as such it might sit for a bit because I need to get back to some existing projects.

**Comment by fgrehm** [comment]
No problemo!

File: scripts/enterenv [hunk 1]
```diff
@@ -1,40 +1,75 @@
...
-  if [[ ! -d "/data/users/$USER" ]]; then
-    mkdir -p "/data/users/$USER"
-    mkdir -p "/data/users/$USER/envs"
-    mkdir -p "/data/users/$USER/home"
-    cp -R /tmp/data/root "/data/users/$USER/root"
+  if [[ ! -d "$USER_ROOT" ]]; then
+    mkdir -p "$USER_ROOT"
+    mkdir -p "$USER_ROOT/envs"
+    if $(wget --spider $GH_ENVY_HOME &>/dev/null); then
+      git clone $GH_ENVY_HOME "$USER_ROOT/home"
```
Comment (fgrehm) [inline scripts/enterenv:48]: Just realized that a `--recurse-submodules` is going to be useful here, as we might want to link our `envy.home`s to our personal dotfiles / vimfiles for example. I'm going to experiment with that and will update the PR accordingly.

Comment (progrium) [inline scripts/enterenv:48]: Mehhhhh I don't want to deal with submodules. :\

Comment (fgrehm) [inline scripts/enterenv:48]: No worries :)

What about adding support for running a setup script from the users home after cloning it for the first time so everyone can wire things up as wanted? (not necessarily for v0.1.0, maybe something for the future)

Comment (progrium) [inline scripts/enterenv:48]: Inside or outside the session container?

Comment (fgrehm) [inline scripts/enterenv:48]: Outside I guess? Possibly just exposing the `HOME` dir to that container. Something like:

``` sh
if [ -x "$USER_ROOT/home/bootstrap.sh" ]; then
  docker run -it --rm \
        --name "$USER.bootstrap" \
        --volume "$HOST_DATA/users/$USER/home:/home/$USER" \
        "$ENV_IMAGE" /home/$USER/setup.sh
fi
```

right below this `git clone`

Comment (progrium) [inline scripts/enterenv:48]: Cause we have the tools for a plugin model, but I just want to wait until we have clear ideas for use. Dokku got out of control really fast. Good and bad.

Anyway, seems like you could do this from within the container via startup scripts. Esp once we have the /root startup script load your user startup script.

Comment (fgrehm) [inline scripts/enterenv:48]: Cool, tks for the insight!

**Comment by fgrehm** [comment]
@progrium are you still interested in any of this? if you are I can try bringing this up to speed with master and squashing some commits.

**Comment by progrium** [comment]
I'm already having trouble modeling some of this in bash cleanly so I'm
planning to start porting most of enterenv to go. So thanks for the PoC but
this PR isn't going to make it.

On Monday, July 6, 2015, Fabio Rehm notifications@github.com wrote:

> @progrium https://github.com/progrium are you still interested in any
> of this? if you are I can try bringing this up to speed with master and
> squashing some commits.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/progrium/envy/pull/25#issuecomment-118852604.

## 

Jeff Lindsay
http://progrium.com

**Comment by fgrehm** [comment]
no worries!

**Pull request closed without merging**